### PR TITLE
Bump metal3 baremetal-operator to v0.5.0

### PR DESCRIPTION
**Pull request by rabi**
This requires controller-runtime related changes for webhooks and SetupWithManager().

### CODE REVIEW

File: api/go.mod [hunk 1]
```diff
@@ -4,18 +4,18 @@ go 1.18
...
-	github.com/metal3-io/baremetal-operator/apis v0.3.1
+	github.com/metal3-io/baremetal-operator/apis v0.4.0
 	github.com/onsi/ginkgo/v2 v2.11.0
 	github.com/onsi/gomega v1.27.8
 	github.com/openstack-k8s-operators/lib-common/modules/common v0.1.0
-	k8s.io/api v0.26.7
-	k8s.io/apimachinery v0.26.7
-	sigs.k8s.io/controller-runtime v0.14.6
+	k8s.io/api v0.27.2
+	k8s.io/apimachinery v0.27.2
```
Comment (abays) [inline api/go.mod:12]: I think we wanted to keep `k8s.io` below `v0.27.0` for now [1].  Maybe it's okay to bump that constraint now though?

[1] https://github.com/openstack-k8s-operators/renovate-config/blob/6b1328bdf0f4697a500288cc4ce2f8b86a87a0b2/default.json#L29-L39

Comment (rabi) [reply api/go.mod:12]: Yeah we have to I guess. Also, looks like there is no easy way other than force merging this and similar changes to other operators that bumps controller-runtime. We change the API go.mod and when we build the meta-operator with the change, it won't work with other operators[1].

Do you've any better idea? 

[1] https://logserver.rdoproject.org/73/73/765d4b1cf98d84ee4fb90680363cb5a48b81c33b/github-check/openstack-baremetal-operator-content-provider/3938978/ci-framework-data/logs/ci_make_009_openstack-operator_-_call_docker-build.log

Comment (kajinamit) [reply api/go.mod:12]: I'm afraid we don't have any better way to bump `k8s.io` to `v0.27.0`, and we may need to force-merge bump with interface fix at the same time.

The past agreement is that we'll bump it when operator-sdk bumps `k8s.io` in its dependencies but that bump has not yet happened so we probably have to hold this now.

Comment (rabi) [reply api/go.mod:12]: Not sure we can fall behind with metal3 waiting for operatorator-sdk to be bumped. We need to force merge core kubernetes libraries (k8s.io/sigs.k8s.io) bump in all operators and then bump openstack-operator to make things green. We know what changes are required in every-operator.

What happened to the plan to move apis to separate repo or something?

**Pull request merged**